gmso/core/topology.py

```diff
@@ -610,6 +631,59 @@ def update_topology(self):
         self.update_connection_types()
         self.is_typed(updated=True)
 
+    def delete_site(self, site):
+        """Remove `site` from the topology
+
+        This method takes a gmso.Site object `site` and
+        removes the site if it exists in the topology.
+
+        Parameters
+        ----------
+        site : gmso.Site
+            A Site object that exists on the topology.
+
+        Raises
+        -------
+        GMSOError
+            If `site` doesn't exist in the topology's sites.
+
+        Notes
+        -----
+        Removing a site from the topology removes any bond,
+        angles, dihedrals and impropers associated with the site.
+        """
+        if site not in self._sites:
+            raise GMSOError(f"Site {site} does not exist in the topology")
+        else:
+            self._sites.discard(site)
+            self._dead_sites.add(site)
+            for connection in site.connections:
+                if connection in self._connections:
+                    self._connections.discard(connection)
+                    self._dead_connections.add(connection)
+                    if isinstance(connection, Bond):
+                        self._bonds.discard(connection)
+                        self._dead_bonds.add(connection)
+                    if isinstance(connection, Angle):
+                        self._angles.discard(connection)
+                        self._dead_angles.add(connection)
+                    if isinstance(connection, Dihedral):
+                        self._dihedrals.discard(connection)
+                        self._dead_dihedrals.add(connection)
+                    if isinstance(connection, Improper):
+                        self._dihedrals.discard(connection)
+                        self._dead_impropers.add(connection)
+                    for partner in connection.connection_members:
+                        if partner is not site:
+                            partner.connections.discard(connection)
+        site.connections.clear()
+
+    def _compact_set(self, member, member_type='sites'):
+        current_dead, current_set = self._del_ref_sets[member_type]
+        if member in current_dead:
+            current_set._compact()
+            current_dead.discard(member)
+
```

Comment:
I feel like a `top.update_top()` is needed somewhere in here to clean up `ConnectionType` objects

Reply:
There's an open PR to get Associations for a `Potential` or an `AtomType` (#378). That PR when merged would deal with removing them more efficiently IMO.